src/functions/indGraphFat.jl

```diff
@@ -0,0 +1,58 @@
+# This implements the variant when A is dense and m < n
+
+struct IndGraphFat{T <: RealOrComplex} <: IndGraph
+  m::Int
+  n::Int
+  A::AbstractArray{T,2}
+  AA::AbstractArray{T,2}
+  F::Base.LinAlg.Cholesky #LDL factorization
+end
+
+function IndGraphFat(A::AbstractArray{T,2}) where {T <: RealOrComplex}
+  m, n = size(A)
+  AA = A * A'
+
+  F = LinAlg.cholfact(eye(m) + AA)
+  #normrows = vec(sqrt.(sum(abs2.(A), 2)))
+
+  IndGraphFat(m, n, A, AA, F)
+end
+
+# is_convex(f::IndGraph) = true
+# is_set(f::IndGraph) = true
+# is_cone(f::IndGraph) = false
+
+function prox!{T <: RealOrComplex}(
+    x::AbstractArray{T, 1},
+    y::AbstractArray{T, 1},
+    f::IndGraphFat,
+    c::AbstractArray{T, 1},
+    d::AbstractArray{T, 1})
+  y[:] = f.F \ (f.A * c + f.AA * d)
```

Comment:
There is no efficient implementation for doing matrix-vector multiplication and adding the result to a vector, only writing it to a vector, so the line `y[:] .+= f.AA * d` will do something like
```julia
tmp = f.AA * d
y[:] .+= tmp
```
However, if we already allocate tmp and store it in `IndGraph`, julia won't have to allocate it every time `prox!` is called, it is only allocated when `IndGraph` is created and then reused. Wherever I wrote `tmp` in my code, it should be `f.tmp` instead.